types/resource.go

```diff
@@ -282,3 +282,51 @@ type BackupStatus struct {
 	BackupError  string `json:"backupError,omitempty"`
 	SnapshotName string `json:"snapshotName"`
 }
+
+type InstanceType string
+
+const (
+	InstanceTypeEngine  = InstanceType("engine")
+	InstanceTypeReplica = InstanceType("replica")
+)
+
+type InstanceManagerState string
+
+const (
+	InstanceManagerStateError    = InstanceManagerState("error")
+	InstanceManagerStateRunning  = InstanceManagerState("running")
+	InstanceManagerStateStopped  = InstanceManagerState("stopped")
+	InstanceManagerStateStarting = InstanceManagerState("starting")
+	InstanceManagerStateUnknown  = InstanceManagerState("unknown")
+)
+
+type InstanceManagerType string
+
+const (
+	InstanceManagerTypeEngine  = InstanceManagerType("engine")
+	InstanceManagerTypeReplica = InstanceManagerType("replica")
+)
+
+type InstanceManagerSpec struct {
+	BootNodeID  string `json:"bootNodeID"`
```

Comment:
Rename it to `NodeBootID`. And should be it in here rather than `InstanceManagerStatus`?

Reply:
Will fix.